File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 6/12/2025, 8:28:01 PM
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i=0;
        vector<int>x;
        vector<int>y;
        while(i<nums.size())
        {
            if(nums[i]!=0)
            {
                x.push_back(nums[i]);
            }
            else{
            y.push_back(nums[i]);
            }
            i++;
        }
        
        nums.assign(x.begin(),x.end());
        nums.insert(nums.end(),y.begin(),y.end());
    }
};

File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 6/12/2025, 8:28:05 PM
class Solution(object):
    def moveZeroes(self, nums):
        j=0
        for num in nums:
            if num!=0:
                nums[j]=num
                j+=1
        for i in range(j,len(nums)):
            nums[i]=0
            
                
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        